File: tests/domains/files/test_list.py
from domains import Runtime
from domains.files.delete import delete_file
from domains.files.listing import list_files
from tests.helpers import unwrap


def test_list_and_filter(runtime: Runtime) -> None:
    root = runtime.workspace.root
    (root / "a.mp4").write_bytes(b"1")
    (root / "sub").mkdir()
    (root / "sub" / "b.wav").write_bytes(b"22")
    (root / "notes.txt").write_bytes(b"333")

    everything = unwrap(list_files(runtime))
    assert [f["path"] for f in everything["files"]] == ["a.mp4", "notes.txt", "sub/b.wav"]
    assert everything["count"] == 3

    only_audio = unwrap(list_files(runtime, kind="audio"))
    assert [f["path"] for f in only_audio["files"]] == ["sub/b.wav"]

    shallow = unwrap(list_files(runtime, recursive=False))
    assert shallow["count"] == 2


def test_list_missing_dir(runtime: Runtime) -> None:
    assert list_files(runtime, "nope").get("code") == "not_found"


def test_delete(runtime: Runtime) -> None:
    target = runtime.workspace.root / "tmp.mp4"
    target.write_bytes(b"xyz")
    result = delete_file(runtime, "tmp.mp4")
    assert result.get("freed_bytes") == 3
    assert not target.exists()
    assert delete_file(runtime, "tmp.mp4").get("code") == "not_found"

File: version-in-rust/tests/domains/files/test_list.rs
use mcp_tools::core::errors::ErrorCode;
use mcp_tools::domains::files::delete::delete_file;
use mcp_tools::domains::files::listing::{list_files, FileKind};

use crate::conftest::runtime;

#[test]
fn test_list_and_filter() {
    let rt = runtime();
    let root = rt.root().to_path_buf();
    std::fs::write(root.join("a.mp4"), b"1").unwrap();
    std::fs::create_dir(root.join("sub")).unwrap();
    std::fs::write(root.join("sub").join("b.wav"), b"22").unwrap();
    std::fs::write(root.join("notes.txt"), b"333").unwrap();

    let everything = list_files(&rt.runtime, ".", None, true).unwrap();
    let paths: Vec<&str> = everything.files.iter().map(|f| f.path.as_str()).collect();
    assert_eq!(paths, ["a.mp4", "notes.txt", "sub/b.wav"]);
    assert_eq!(everything.count, 3);

    let only_audio = list_files(&rt.runtime, ".", Some(FileKind::Audio), true).unwrap();
    let paths: Vec<&str> = only_audio.files.iter().map(|f| f.path.as_str()).collect();
    assert_eq!(paths, ["sub/b.wav"]);

    let shallow = list_files(&rt.runtime, ".", None, false).unwrap();
    assert_eq!(shallow.count, 2);
}

#[test]
fn test_list_missing_dir() {
    let rt = runtime();
    let error = list_files(&rt.runtime, "nope", None, true).unwrap_err();
    assert_eq!(error.code, ErrorCode::NotFound);
}

#[test]
fn test_delete() {
    let rt = runtime();
    let target = rt.root().join("tmp.mp4");
    std::fs::write(&target, b"xyz").unwrap();
    let result = delete_file(&rt.runtime, "tmp.mp4").unwrap();
    assert_eq!(result.freed_bytes, 3);
    assert!(!target.exists());
    let error = delete_file(&rt.runtime, "tmp.mp4").unwrap_err();
    assert_eq!(error.code, ErrorCode::NotFound);
}
